LibGAP.jl/src/gap_to_julia.jl

```diff
@@ -121,7 +122,7 @@ function gap_to_julia( ::Type{Array{Obj,1}}, obj :: MPtr , recursion_dict = IdDi
     new_array = Array{Any,1}( undef, len_list)
     recursion_dict[obj] = new_array
     for i in 1:len_list
-        current_obj = obj[i]
+        current_obj = ElmList(obj,i)
```

Comment:
Hm. Should we then (or even instead) also adjust `GAP.Globals.ELM_LIST` to use `ElmList` (for performance, and to also return `nothing`.

And perhaps we should also use `GAP_AssList` for `Base.setindex!` instead of `GAP.Globals.ASS_LIST`>

Reply:
I am not completely sure about this. `ELM_LIST` behaves different, as it returns `fail` instead of `nothing`. We currently do not map `fail` to `nothing`, which is probably the right behaviour. I would try to mimic GAP's behaviour here, which is the behaviour of `ELM_LIST` (or even throwing an error).

Reply:
First off, sorry, I meant to write: "... should we adjust `Base.getindex` to use `ElmLst` instead of `GAP.Globals.ELM_LIST` ..."

Secondly, I am not aware of any variant of `ELM_LIST` returning `fail`, what are you referring to exactly? I am seeing this:
```
julia> xx = GAP.EvalString("[1,,3]")
GAP: [ 1, , 3 ]

julia> GAP.Globals.ELM_LIST(xx,2)
Error, List Element: <list>[2] must have an assigned value
ERROR: Error thrown by GAP
```

Reply:
I think it's fine to have an error here. What is less clear to me is whether we want to rely on GAP throwing that error. We could instead write a slightly more elaborate getter, which uses `ElmList`, and then if `nothing` is returned, throws an error.

Anyway, this is not a blocker for this PR.